irony-cdb.el

```diff
@@ -114,6 +128,24 @@ for files that it cannot handle."
 ;; Functions
 ;;
 
+(defun irony-cdb--locate-dominating-file-with-dirs (file name subdirectories)
+  "Convenience wrapper around `locate-dominating-file'
+
+Looks up the directory hierarchy from FILE for to locate any directory
+in `subdirectories` which contains NAME. Stops at the first parent
+directory where the test succeeds and returns the full path to file
+found."
+  (let ((check-subdirs
```

Comment:
I don't understand the need of this `check-subdirs` variable, can't you just loop over the directories directly?